### ISSUE TITLE
Add type annotations

### ISSUE THREAD

**Issue by sebastinas** [bpython/bpython#892]
Some parts of the code are already annotated with types. Let's add to the remaining parts as well.

**Comment by Mayur619** [comment]
Hi @sebastinas, I would like to work on this

**Comment by sebastinas** [comment]
@Mayur619 Feel free to start with any file that does not already have type annotations. Easiest is probably to start with a module that only imports stuff from the standard library. For others,  we probably need stubs or type annotations in our dependencies.

**Comment by supremestdoggo** [comment]
Hi! Can I help?

**Comment by victoroliveirab** [comment]
Hey, @sebastinas, are there any more files needing help?

**Comment by sebastinas** [comment]
@victoroliveirab  Yes, there are. You can start by running `mypy` on the `bpython` source tree and start from there.

**Comment by Ben-Reg** [comment]
Hi - I'm new and this looks like a good place to start.  Can I start by working on formatter.py?

**Comment by thomasballinger** [comment]
@Ben-Reg that sounds good, thanks for taking a shot at this!

**Comment by Ben-Reg** [comment]
@thomasballinger, if I have questions, is this the best place to post them?  One of the variables in formatter.py is used in a way that makes assigning a type difficult.  I'm unsure whether I should rewrite it to allow for stricter typing, or just use the Any type.

**Comment by thomasballinger** [comment]
@Ben-Reg opening an in-progress pull request might be the best way to ask since the context will be right there, but you can ask here too.

Which variable is it? Even though the file is short I'm having trouble guessing, so I can't suggest something specific. Starting with any sounds good to me, and we can discuss more as code review.

**Comment by Ben-Reg** [comment]
Thanks @thomasballinger, I opened pull request #947 to add the context.  The issue is with the line `token = token.parent`

**Comment by spotted-cat** [comment]
Heya I would like to help!? @sebastinas

**Comment by thomasballinger** [comment]
@spotted-cat Great! The process is described above, do you have any questions about it? Go right ahead, looking forward to a pull request.

**Comment by RCReddyN** [comment]
Hi! I would like to help.
